vertx-web/src/main/java/io/vertx/ext/web/handler/SessionHandler.java

```diff
@@ -220,6 +221,18 @@ static SessionHandler create(SessionStore sessionStore) {
 	@Fluent
   SessionHandler flush(RoutingContext ctx, Handler<AsyncResult<Void>> handler);
 
+  /**
+   * Flush a context session earlier to the store, this will allow the end user to have full control on the event of
+   * a failure at the store level. Once a session is flushed no automatic save will be performed at end of request.
+   *
+   * @param ctx the current context
+   * @param ignoreStatus flush regardless of response status code
```

Comment:
I'm curious here, what is the reason behind this flag? When would we want to save a session when it may or may not result in a request error? I'd assume that if the request is known to error, probably we don't want to save anything to reduce the chance of ending up in a bad state.

Reply:
This is basically used for force saving a session in the middle of a sockjs connection which may be potentially long lived. In cases where we are not relying on response status codes for errors we can use this. For example we may be sending errors as messages over the sockjs channel without closing the channel at all, and/or need to save the session while the channel is still open.